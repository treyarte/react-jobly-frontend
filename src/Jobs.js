import React, { useState, useEffect } from 'react';
import { Redirect } from 'react-router-dom';
import JobCard from './JobCard';
import JoblyAPI from './JoblyAPI';
import SearchBar from './SearchBar';

const Jobs = ({ checkLogin }) => {
  const [jobs, setJobs] = useState([]);

  const handleSetJobs = async (searchParams) => {
    const jobsArr = await JoblyAPI.getJobs(searchParams);
    setJobs(jobsArr);
  };

  useEffect(() => {
    async function getJobs() {
      const jobsArr = await JoblyAPI.getJobs();
      setJobs(jobsArr);
    }

    getJobs();
  }, []);

  if (!checkLogin()) return <Redirect to='/login' />;

  return (
    <>
      <div className='jobs col-md-8 offset-md-2'>
        <div>
          <SearchBar handleData={handleSetJobs} />
        </div>
        {jobs.length !== 0 ? (
          <div>
            {jobs.map((job) => (
              <JobCard key={job.id} job={job} />
            ))}
          </div>
        ) : (
          <div>
            <p>Sorry no results found!</p>
          </div>
        )}
      </div>
    </>
  );
};

export default Jobs;
