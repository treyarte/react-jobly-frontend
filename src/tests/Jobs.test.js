import React from 'react';
import Jobs from '../Jobs';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';

describe('Jobs Component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Jobs loggedIn={true} />);

    expect(asFragment).toMatchSnapshot();
  });

  it('Should render a searchBar', () => {
    const wrapper = shallow(<Jobs />);

    const searchBar = wrapper.find('SearchBar');

    expect(searchBar.length).toEqual(1);
  });
});
