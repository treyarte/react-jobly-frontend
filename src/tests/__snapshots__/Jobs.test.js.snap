// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jobs Component should match snapshot 1`] = `[Function]`;
